sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala

```diff
@@ -1115,6 +1115,69 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
     checkAnswer(df8, Seq(Row("alex")))
   }
 
+  test("scan with filter push-down with misc functions") {
+    Seq(false, true).foreach { ansiMode =>
+      withSQLConf(SQLConf.ANSI_ENABLED.key -> ansiMode.toString) {
```

Comment:
Why we need test `ANSI_ENABLED` ?

Reply:
Because MD5 and SHA functions need to convert name to binary type, cast will be added.
So we need test ANSI_ENABLED.

Reply:
what's the difference between ansi on and off here?

Reply:
The datasource supports pushdown CAST only if ansi is on.

Reply:
The datasource supports pushdown CAST only if ansi is on.

Reply:
Can we avoid the cast? Let's add a binary type column to the testing table.

Reply:
OK